src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.task.Task;
+
+/**
+ * Represent storage for tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList.
+     * Creates a TaskList with an empty ArrayList of Tasks.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor for TaskList.
+     * Creates a TaskList containing the tasks input.
+     *
+     * @param tasks ArrayList of Tasks to initialize TaskList with.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns the count of Tasks stored.
+     *
+     * @return count of Tasks stored.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Returns Task at index.
+     * Index is 0-based.
+     *
+     * @param idx Index of Task to be returned.
+     * @return Task stored at index.
+     */
+    public Task get(int idx) {
+        return this.tasks.get(idx);
+    }
+
+    /**
+     * Adds Task to TaskList.
+     *
+     * @param task Task to be added.
+     * @return The added Task.
+     */
+    public Task add(Task task) {
+        this.tasks.add(task);
+        return task;
+    }
+
+    /**
+     * Removes Task from TaskList.
+     *
+     * @param idx Index of Task to be removed.
+     * @return The removed Task.
+     */
+    public Task remove(int idx) {
+        Task tempTask = this.tasks.get(idx);
```

Comment:
Maybe could rename this variable to taskToBeRemoved?

Reply:
Thanks for the suggestion! That is indeed a more descriptive name. I will make the changes in the next push.